src/fidesops/service/masking/strategy/masking_strategy_factory.py

```diff
@@ -10,70 +10,57 @@
     MaskingConfiguration,
 )
 from fidesops.service.masking.strategy.masking_strategy import MaskingStrategy
-from fidesops.service.masking.strategy.masking_strategy_aes_encrypt import (
-    AesEncryptionMaskingStrategy,
-)
-from fidesops.service.masking.strategy.masking_strategy_hash import HashMaskingStrategy
-from fidesops.service.masking.strategy.masking_strategy_hmac import HmacMaskingStrategy
-from fidesops.service.masking.strategy.masking_strategy_nullify import (
-    NullMaskingStrategy,
-)
-from fidesops.service.masking.strategy.masking_strategy_random_string_rewrite import (
-    RandomStringRewriteMaskingStrategy,
-)
-from fidesops.service.masking.strategy.masking_strategy_string_rewrite import (
-    StringRewriteMaskingStrategy,
-)
 
+logger = logging.getLogger(__name__)
 
-class SupportedMaskingStrategies(Enum):
-    """
-    The supported methods by which Fidesops can pseudonymize data.
-    """
 
-    string_rewrite = StringRewriteMaskingStrategy
-    random_string_rewrite = RandomStringRewriteMaskingStrategy
-    null_rewrite = NullMaskingStrategy
-    hash = HashMaskingStrategy
-    aes_encrypt = AesEncryptionMaskingStrategy
-    hmac = HmacMaskingStrategy
+class MaskingStrategyFactory:
+    registry: Dict[str, MaskingStrategy] = {}
+    valid_strategies = ""
 
     @classmethod
-    def __contains__(cls, item: str) -> bool:
-        try:
-            cls[item]
-        except KeyError:
-            return False
+    def register(cls, name: str) -> Callable[[MaskingStrategy], MaskingStrategy]:
+        def wrapper(strategy_class: MaskingStrategy) -> MaskingStrategy:
+            if name in cls.registry:
+                logger.warning(
+                    f"Strategy with name {name} already exists in MaskingStrategy registry. It referred to class {cls.registry[name]}, but will now refer to {strategy_class}"
+                )
+            logger.info(
+                f"Registering MaskingStrategy class {strategy_class} under name {name}"
+            )
+            cls.registry[name] = strategy_class
+            cls.valid_strategies = ", ".join(cls.registry.keys())
+            return strategy_class
```

Comment:
Just re-ordering and cleaning up the logging here, including downgrading the registration log line to the debug-level. I think these changes make the code more intuitive. Also, as a nit, instead of reflecting the `strategy_class` that was passed, we should return the new value of `cls.registry[name]`:
```suggestion
        def wrapper(strategy_class: MaskingStrategy) -> MaskingStrategy:
            logger.debug(
                f"Registering new masking strategy '{strategy_class}' under name '{name}'"
            )

            if name in cls.registry:
                logger.warning(
                    f"Masking strategy with name '{name}' already exists. It previously referred to class '{cls.registry[name]}', but will now refer to '{strategy_class}'"
                )

            cls.registry[name] = strategy_class
            cls.valid_strategies = ", ".join(cls.registry.keys())
            return cls.registry[name]
```

Reply:
i like it. definitely easier to follow.